nft_ingester/src/program_transformers/bubblegum/mint_v1.rs

```diff
@@ -211,121 +164,36 @@ where
-                            creator: Set(c.address.to_bytes().to_vec()),
-                            verified: Set(c.verified),
-                            seq: Set(Some(seq as i64)),
-                            ..Default::default()
-                        });
-
-                        creators_set.insert(c.address);
-                    }
-
-                    // This statement will update base information for each creator.
-                    let query = asset_creators::Entity::insert_many(db_creator_infos)
-                        .on_conflict(
-                            OnConflict::columns([
-                                asset_creators::Column::AssetId,
-                                asset_creators::Column::Creator,
-                            ])
-                            .update_columns([
-                                asset_creators::Column::Position,
-                                asset_creators::Column::Share,
-                                asset_creators::Column::SlotUpdated,
-                            ])
-                            .to_owned(),
-                        )
-                        .build(DbBackend::Postgres);
-                    txn.execute(query).await?;
+                // Upsert into `asset_v1_account_attachments` table.
+                let (edition_attachment_address, _) = find_master_edition_account(&id);
+                unprotected_upsert_asset_v1_account_attachments(
+                    txn,
+                    edition_attachment_address.to_bytes().to_vec(),
+                    slot_i,
+                )
+                .await?;
 
-                    // This statement will update whether the creator is verified and the `seq`
-                    // number.  `seq` is used to protect the `verified` field, allowing for `mint`
-                    // and `verifyCreator` to be processed out of order.
-                    let mut query = asset_creators::Entity::insert_many(db_creator_verified_infos)
-                        .on_conflict(
-                            OnConflict::columns([
-                                asset_creators::Column::AssetId,
-                                asset_creators::Column::Creator,
-                            ])
-                            .update_columns([
-                                asset_creators::Column::Verified,
-                                asset_creators::Column::Seq,
-                            ])
-                            .to_owned(),
-                        )
-                        .build(DbBackend::Postgres);
-                    query.sql = format!(
-                        "{} WHERE excluded.seq > asset_creators.seq OR asset_creators.seq IS NULL",
-                        query.sql
-                    );
-                    txn.execute(query).await?;
-                }
+                // Upsert into `asset_creators` table.
+                unprotected_upsert_creators(
```

Comment:
I don't think we can do an unprotected upsert here. What happens if there's an out-of-order update with creator verify/unverify?

Reply:
The naming `unprotected_upsert_creators` is actually a slight misnomer, as it actually had some protection inside it for the `verified` flag which you noticed and mentioned in later comment.

Example:
There's two creators, A and B.

`verify_creator` at `seq` 2 indexed first, and it inserts a row for creator B that only contains the creator address, `verified` set to true, `verifiedSeq` set to 2.

`mint_v1` at `seq` 1 is indexed next and inserts the row for creator A, but for creator B only updates the record for `postion`, `share`, and `slotUpdated`.   It leaves the `verified` flag as-is because of `verifiedSeq`.

If there happened to be an `update_metadata` afterwards that changed creator positions and verification status, it would have blocked both the `verify_creator` and the `mint_v1` due to the blanket protection from `update_metadata_seq`.

However, this all needs to be revisited anyways as you have shown that the blanket protection from `update_metadata_seq` is unsound approach with concurrent ingestor processes.

Reply:
Fixed in https://github.com/metaplex-foundation/digital-asset-rpc-infrastructure/pull/134